File: Java/TestWebScraping/src/main/java/org/example/WebScraping.java
package org.example;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.net.URL;
import java.nio.file.*;
import java.util.zip.ZipEntry;
import java.util.zip.ZipOutputStream;

public class WebScraping {

    public static void main(String[] args) {

        String baseUrl = "https://www.gov.br/ans/pt-br/acesso-a-informacao/participacao-da-sociedade/atualizacao-do-rol-de-procedimentos";

        String userHome = System.getProperty("user.home") + "\\Desktop\\intuitive-care\\intuitive-care-Teste\\Java\\TestWebScraping";

        String downloadDir = userHome + "\\downloads";
        String zipFilePath = userHome + "\\anexos.zip";

        try {

            Path downloadPath = Paths.get(downloadDir);
            Files.createDirectories(downloadPath);
            System.out.println("Diretório de downloads: " + downloadPath.toAbsolutePath());


            Document document = Jsoup.connect(baseUrl).get();


            Elements links = document.select("a[href]");


            for (Element link : links) {
                String href = link.absUrl("href");
                if ((href.contains("Anexo_I") || href.contains("Anexo_II")) && href.toLowerCase().endsWith(".pdf")) {
                    String fileName = href.substring(href.lastIndexOf("/") + 1);
                    downloadFile(href, Paths.get(downloadDir, fileName).toString());
                }
            }


            zipFiles(downloadDir, zipFilePath);

            System.out.println("Download e compactação concluídos com sucesso!");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }


    private static void downloadFile(String fileUrl, String destination) throws IOException {
        Files.createDirectories(Paths.get(destination).getParent()); // Garantir que o diretório exista
        try (InputStream in = new URL(fileUrl).openStream()) {
            Files.copy(in, Paths.get(destination), StandardCopyOption.REPLACE_EXISTING);
            System.out.println("Arquivo baixado: " + destination);
        }
    }


    private static void zipFiles(String sourceDir, String zipFilePath) throws IOException {
        Path sourcePath = Paths.get(sourceDir);


        if (!Files.exists(sourcePath)) {
            System.err.println("Diretório não encontrado: " + sourcePath.toAbsolutePath());
            return;
        }

        try (ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(zipFilePath))) {
            Files.walk(sourcePath)
                    .filter(Files::isRegularFile)
                    .forEach(filePath -> {
                        try {
                            ZipEntry zipEntry = new ZipEntry(filePath.getFileName().toString());
                            zos.putNextEntry(zipEntry);
                            Files.copy(filePath, zos);
                            zos.closeEntry();
                        } catch (IOException e) {
                            e.printStackTrace();
                        }
                    });
        }

        System.out.println("Arquivos compactados em: " + zipFilePath);
    }
}

File: Python/testeWebscraping/web_scraping.py
import os
import requests
from bs4 import BeautifulSoup
import zipfile

base_dir = os.getcwd() 
download_dir = os.path.join(base_dir, 'anexos')  

def download():
    for url in all_urls:
        try:
            if 'href' in url.attrs and ('Anexo_I' in url['href'] or 'Anexo_II' in url['href']) and url['href'].endswith('.pdf'):
                file_url = url['href']
                file_name = file_url.split('/')[-1]
                
                os.makedirs(download_dir, exist_ok=True)

                file_response = requests.get(file_url)

                with open(os.path.join(download_dir, file_name), 'wb') as f:
                    f.write(file_response.content)
                print(f"Arquivo baixado: {file_name}")
        except Exception as e:
            print(f"Erro ao baixar o arquivo: {e}")

def zip_directory(folder_path, zip_path):
    with zipfile.ZipFile(zip_path, mode='w') as zipf:
        len_dir_path = len(folder_path)
        for root, _, files in os.walk(folder_path):
            for file in files:
                file_path = os.path.join(root, file)
                zipf.write(file_path, file_path[len_dir_path:])
    print(f"Arquivos comprimidos em: {zip_path}")

url = 'https://www.gov.br/ans/pt-br/acesso-a-informacao/participacao-da-sociedade/atualizacao-do-rol-de-procedimentos'
response = requests.get(url)
content = BeautifulSoup(response.text, 'html.parser')
all_urls = content.find_all('a')

download()

zip_directory(download_dir, os.path.join(base_dir, 'anexos-zip.zip'))
